DOC: Improve code example for DataFrame.join

ChristofKaufmann opened the pull request:
The modified code example does not have unique values in the 'key' column in the calling DataFrame `df`. With unique keys the special behavior of `join` has not been highlighted, since it just added a new column. With the modified values the example shows that the 'key' column is really used as key.

ChristofKaufmann committed:
DOC: Improve code example for DataFrame.join

The modified code example does not have unique values in the 'key' column in the calling DataFrame `df`. With unique keys the special behavior of `join` has not been highlighted, since it just added a new column. With the modified values the example shows that the 'key' column is really used as key.

phofl reviewed (changes requested):
Could you keep the original example and add a new one instead?

ChristofKaufmann commented:
OK, should I add it for all calls or just with the last and maybe most interesting call, i. e. `df.join(other.set_index('key'), on='key')`?

phofl commented:
I think the last one is enough

ChristofKaufmann committed:
Use the modified values as additional example

This adds back the original example and adds the new example as an additional one.

pep8speaks commented:
Hello @ChristofKaufmann! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/core/frame.py`](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py):

> [Line 9129:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9129): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9138:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9138): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9141:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9141): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9147:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9147): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9149:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9149): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9158:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9158): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9162:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9162): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9172:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9172): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9177:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9177): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9188:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9188): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9191:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9191): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 9200:1](https://github.com/pandas-dev/pandas/blob/932e8f5016aaf0985ff942d9aff9107d27f85608/pandas/core/frame.py#L9200): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

ChristofKaufmann committed:
Remove whitespace on blank lines

phofl reviewed (approved):
lgtm

jreback commented:
thanks @ChristofKaufmann

ChristofKaufmann commented:
Thank you!
